File: artemis-entity-tracker/src/net/namekdev/entity_tracker/connectors/WorldUpdateListener.java
package net.namekdev.entity_tracker.connectors;

import java.util.BitSet;

/**
 *
 * @author Namek
 */
public interface WorldUpdateListener {
	public static final int ENTITY_ADDED = 1 << 1;
	public static final int ENTITY_DELETED = 1 << 2;
//	public static final int CHANGED = 1 << 3;
	public static final int ENTITY_SYSTEM_STATS = 1 << 4;


	int getListeningBitset();

	void addedSystem(int index, String name, BitSet allTypes, BitSet oneTypes, BitSet notTypes);

	void addedManager(String name);

	void addedComponentType(int index, String name);

	void updatedEntitySystem(int index, int entitiesCount, int maxEntitiesCount);

	void addedEntity(int entityId, BitSet components);

//	void changed(Entity e);

	void deletedEntity(int entityId);


}

File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/connectors/WorldUpdateListener.kt
package net.namekdev.entity_tracker.connectors

import com.artemis.utils.BitVector;
import net.namekdev.entity_tracker.model.ComponentTypeInfo

/**
 *
 * @author Namek
 */
interface WorldUpdateListener {
	fun injectWorldController(controller: WorldController)
	val listeningBitset: Int
	fun addedSystem(index: Int, name: String, allTypes: BitVector?, oneTypes: BitVector?, notTypes: BitVector?)
	fun addedManager(name: String)
	fun addedComponentType(index: Int, info: ComponentTypeInfo)
	fun updatedEntitySystem(index: Int, entitiesCount: Int, maxEntitiesCount: Int)
	fun addedEntity(entityId: Int, components: BitVector)
	//	void changed(Entity e);
	fun deletedEntity(entityId: Int)

	fun updatedComponentState(entityId: Int, componentIndex: Int, valueTree: Any)

	companion object {
		const val ENTITY_ADDED = 1 shl 1
		const val ENTITY_DELETED = 1 shl 2
		//	const val CHANGED = 1 shl 3;
		const val ENTITY_SYSTEM_STATS = 1 shl 4
	}
}